### ISSUE TITLE
caravan error

### ISSUE THREAD

**Issue by TinyGecko** [jptrrs/BurnItForFuel#4]
When one of my caravans arrived at another settlement and I tried to trade with them, the trade window had its elements placed in the wrong positions and I was not able to trade.
I closed the window and now the button for trading with that settlement is gone and I can't interact with them anymore.

Here is the error it produced:

Exception while generating thing set: System.NullReferenceException: Object reference not set to an instance of an object
at BurnItForFuel.CompSelectFuel.IsVehicle () <0x00030>
at BurnItForFuel.CompSelectFuel.SafeToMixFuels () <0x00078>
at BurnItForFuel.CompSelectFuel.Initialize (Verse.CompProperties) <0x00459>
at (wrapper dynamic-method) Verse.ThingWithComps.InitializeComps_Patch1 (object) <0x00133>
at Verse.ThingWithComps.PostMake () <0x00034>
at (wrapper dynamic-method) Verse.ThingMaker.MakeThing_Patch2 (Verse.ThingDef,Verse.ThingDef) <0x003b7>
at RimWorld.StockGeneratorUtility.TryMakeForStockSingle (Verse.ThingDef,int) <0x00222>
at RimWorld.StockGeneratorUtility/<TryMakeForStock>c__Iterator0.MoveNext () <0x0008b>
at RimWorld.StockGenerator_Tag/<GenerateThings>c__Iterator0.MoveNext () <0x002f4>
at RimWorld.ThingSetMaker_TraderStock.Generate (RimWorld.ThingSetMakerParams,System.Collections.Generic.List`1<Verse.Thing>) <0x0035a>
at (wrapper dynamic-method) RimWorld.ThingSetMaker.Generate_Patch1 (object,RimWorld.ThingSetMakerParams) <0x00173>

Verse.Log:Error(String, Boolean)
RimWorld.ThingSetMaker:Generate_Patch1(Object, ThingSetMakerParams)
RimWorld.Planet.SettlementBase_TraderTracker:RegenerateStock()
RimWorld.Planet.SettlementBase_TraderTracker:get_StockListForReading()
RimWorld.Planet.SettlementBase:get_Goods()
RimWorld.TradeDeal:AddAllTradeables()
RimWorld.TradeDeal:Reset()
RimWorld.TradeDeal:.ctor()
RimWorld.TradeSession:SetupWith(ITrader, Pawn, Boolean)
RimWorld.Dialog_Trade:.ctor(Pawn, ITrader, Boolean)
RimWorld.Planet.<TradeCommand>c__AnonStorey0:<>m__0()
Verse.Command_Action:ProcessInput(Event)
Verse.GizmoGridDrawer:DrawGizmoGrid(IEnumerable`1, Single, Gizmo&)
RimWorld.InspectGizmoGrid:DrawInspectGizmoGridFor_Patch0(IEnumerable`1, Gizmo&)
RimWorld.Planet.WorldInspectPane:DrawInspectGizmos()
RimWorld.InspectPaneUtility:ExtraOnGUI(IInspectPane)
RimWorld.Planet.WorldInspectPane:ExtraOnGUI()
Verse.WindowStack:WindowStackOnGUI()
RimWorld.UIRoot_Play:UIRootOnGUI()
Verse.Root:OnGUI()

I play with your mod for a long time now and trade a lot with other settlements. This is the first time I see this error.

**Comment by jptrrs** [comment]
Are you running the latest version?
https://github.com/jptrrs/BurnItForFuel/releases/tag/v1.2.2
If you are, would you mind trying this, so I get more info on this? 
https://steamcommunity.com/workshop/filedetails/discussion/1823276856/1609400247631534303/

**Comment by TinyGecko** [comment]
When it happened I was running 1.2.1, so it was before the two updates you've pushed today.

It won't be easy to reproduce this error, from the hundreds of trades I've made with your mod running, this only happened once. :(
Or it was something that was introduced with 1.2.1, I did only a handful of trades with this version...

I will try to reproduce it and trade as much as possible. If I can find out more I will post it here!

In the meantime do you know if it is possible to reset the corrupted settlement to make trading with them available again?

**Comment by jptrrs** [comment]
> When it happened I was running 1.2.1, so it was before the two updates you've pushed today.
Actually, that was available for some time now, I just haven't marked it as a release (that's not automatic, apparently...)

> Or it was something that was introduced with 1.2.1, I did only a handful of trades with this version...
The function that was failing you was introduced with v. 1.2, so it might be the case. But the same error was showing up in other situations, and it was corrected in 1.2.2. Its quite probable it won't happen on trade anymore as well. If it does, let me know.

> In the meantime do you know if it is possible to reset the corrupted settlement to make trading with them available again?
I have no clue!

**Comment by TinyGecko** [comment]
I'm glad that it is already fixed, thanks for your help!

**Comment by TinyGecko** [comment]
I'm sorry, but it happened again on a new game with the latest version on steam. I've traded many times with a settlement without issues, until suddenly it became corrupted and now I can't interact with them anymore.

**Comment by jptrrs** [comment]
Hi, TinyGecko. Yeah, I do get notified on closed issues, I just haven't been paying attention lately, sorry.
Let's see what we can do about that. Can you please post the error log?
If you need directions for that: https://steamcommunity.com/workshop/filedetails/discussion/1823276856/1609400247631534303/

**Comment by TinyGecko** [comment]
Well that's embarassing now, but the moment it happened I was sure that it was the same error from my initial post again so I did not save it. I know I should have done that, sorry. :( I hoped you wouldn't ask.

On the bright side I was able to fix the broken settlement by searching it in my savegame and replacing the trader info with
`						<trader Class="Settlement_TraderTracker">
							<tmpSavedPawns />
							<stock IsNull="True" />
							<lastStockGenerationTicks>-1</lastStockGenerationTicks>
						</trader>`

Next time I will post the error!
